negotiator.go

```diff
@@ -33,9 +33,11 @@ func (l Negotiator) RoundTrip(req *http.Request) (res *http.Response, err error)
 	if rt == nil {
 		rt = http.DefaultTransport
 	}
+
 	// If it is not basic auth, just round trip the request as usual
-	reqauth := authheader(req.Header.Get("Authorization"))
-	if !reqauth.IsBasic() {
+	reqauth := authheader(req.Header.Values("Authorization"))
+	isBasic, authHeader := reqauth.IsBasic()
```

Comment:
```suggestion
	isBasic, reqauthBasic := reqauth.IsBasic()
```
because it is like the reqauth variable